apps/site-landing/src/containers/BenefitsSimulator/Form/Advanced.tsx

```diff
@@ -0,0 +1,276 @@
+        label={basic.rangeDuration}
+        scaleType="unit"
+        unit="year"
+        maximum={60}
+        initial={years}
+        onChange={dbOnSetYears}
+      />
+      <Accordion>
+        <PeriodicalGroup
+          title={translations.income}
+          index={0}
+          activeIndexState={activeIdxState}
+          params={params.income}
+          setParams={onChangedInObj("income")}
+        />
+        <PeriodicalGroup
+          title={translations.credit}
+          index={1}
+          activeIndexState={activeIdxState}
+          params={params.credit}
+          setParams={onChangedInObj("credit")}
+        >
+          <RangeFieldAndScale
+            label={translations.cashBackRate}
+            scaleType="percent"
+            maximum={0.05}
+            step={0.01}
+            initial={params.credit.cashBackRate}
+            onChange={v => onChangedInObj("credit")({ ...params.credit, cashBackRate: v })}
+          />
+          <RangeFieldAndScale
+            label={translations.interestRate}
+            scaleType="percent"
+            maximum={1}
+            step={0.01}
+            initial={params.credit.interestRate}
+            onChange={v => onChangedInObj("credit")({ ...params.credit, interestRate: v })}
+          />
+
+        </PeriodicalGroup>
+        <PeriodicalGroup
+          title={translations.cash}
+          index={2}
+          activeIndexState={activeIdxState}
+          params={params.cash}
+          setParams={onChangedInObj("cash")}
+        />
+        <ShockAbsorberGroup
+          title={translations.cash}
+          index={3}
+          activeIndexState={activeIdxState}
+          params={params.shockAbsorber}
+          setParams={onChangedInObj("shockAbsorber")}
+        />
+      </Accordion>
+    </div>
+  )
+}
+
+type GroupProps<T> = {
+  title: MessageDescriptor;
+  index: number;
+  activeIndexState: UseNumberState;
+  params: T;
+  setParams: (v: T) => void;
+}
+
+const PeriodicalGroup: React.FC<GroupProps<PeriodicalParams>> = ({ title, index, params, setParams, activeIndexState, children }) => {
+
+  const setNamedValue = (name: keyof PeriodicalParams) =>
+    (val: any) => setParams({
+      ...params,
+      [name]: val,
+    }
+    )
+  const active = activeIndexState[0] === index;
+  const toggleActive = () => activeIndexState[1](
+    activeIndexState[0] == index
+      ? -1
+      : index
+  )
+  return (
+    <>
+      <Accordion.Title
+        active={activeIndexState[0] === index}
+        index={0}
+        onClick={toggleActive}
+      >
+        <Icon name='dropdown' />
+        <FormattedMessage {...title} />
+      </Accordion.Title>
+      <Accordion.Content active={active} >
+        <RangeFieldAndScale
+          label={translations.weekly}
+          scaleType="currency"
+          currency="CAD"
+          maximum={1000}
+          step={1}
+          initial={params.weekly}
+          onChange={setNamedValue("weekly")}
+        />
```

Comment:
Those three range fields could just be one and three radio buttons.

Reply:
I think all 3 should be there.  They mean different things.  Fer example - I (theoretically) get paid weekly, child benefits arrive monthly, and once per year I get a bonus.  Then I buy groceries weekly, I pay rent monthly, and I go on vacation once per year.

It might sound pedantic, but I want this to be as accurate as possible.  Also, these things have surprisingly large effects over time.  Getting paid weekly rather than monthly means maybe $10K more in savings at retirement.

Play with the simulator a bit and you'll see what I mean.  (Also, that it's a huge pain to play with).